refactor: Move project related value objects to admin.domain.model package and cleanup code

seakayone opened the pull request:
<!-- Important! Please follow the guidelines for naming Pull Requests: https://docs.dasch.swiss/latest/developers/dsp/contribution/ -->

## Pull Request Checklist

### Task Description/Number

<!-- Please add either the issue number or, in case of unscheduled work, a short description of the task at hand -->

[from 2.13.2](https://github.com/scala/scala/releases/tag/v2.13.2), the `sealed abstract case class private` can be reduced  by [enabling `-Xsource:3`](https://docs.scala-lang.org/scala3/guides/migration/tooling-tour.html#the-scala-213-compiler). From [scala/scala#7702](https://github.com/scala/scala/pull/7702):

> Backport from dotty:
> 
> * If a case class constructor is `private` or `private[foo]`: the synthesized `copy` and `apply` methods will have the same access modifier.
>  * If a case class constructor is `protected` or `protected[foo]`: the synthesized `copy` method will have the same access modifier. The synthesized `apply` method will remain public, because `protected` doesn't make sense in an object. 
>
> Obviously, if a user defines a custom `copy` or `apply` method, that one—including its access modifier—will take precedence.

see also https://tuleism.github.io/blog/2020/scala-smart-constructors/

This way it is also possible to reduce boxing for [value classes which can extend `AnyVal`](https://docs.scala-lang.org/overviews/core/value-classes.html).

However this PR does not include the activation of that flag as some existing code needs to be updated as well, this is done in https://github.com/dasch-swiss/dsp-api/pull/2924

### PR Type

- [ ] build/chore: maintenance tasks (no production code change)
- [ ] docs: documentation changes (no production code change)
- [ ] feat: represents new features
- [ ] fix: represents bug fixes
- [ ] perf: performance improvements
- [x] refactor: represents production code refactoring
- [ ] test: adding or refactoring tests (no production code change)

### Basic requirements for bug fixes and features

- [x] Tests for the changes have been added
- [ ] Docs have been added / updated

### Does this PR introduce a breaking change?

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->

- [ ] Yes

### Does this PR change client-test-data?

- [ ] Yes

mpro7 commented on webapi/src/test/scala/dsp/valueobjects/KnoraProjectSpec.scala:
```diff
@@ -0,0 +1,182 @@
+    }
+  )
+
+  private val nameTest = suite("Name")(
+    test("pass invalid Name and expect an error to be returned") {
+      val invalidNames =
+        Gen.stringBounded(0, 2)(Gen.printableChar) ++ Gen.stringBounded(257, 300)(Gen.printableChar)
+      check(invalidNames) { name =>
+        assertTrue(
+          Longname.from(name) == Validation.fail(ValidationException("Longname must be 3 to 256 characters long."))
+        )
+      }
+    },
+    test("pass a valid value and successfully create value object") {
+      val validNames = Gen.stringBounded(3, 256)(Gen.printableChar)
+      check(validNames) { name =>
```
```suggestion
  private val nameTest = suite("Longname")(
    test("pass invalid Longname and expect an error to be returned") {
      val invalidLongnames =
        Gen.stringBounded(0, 2)(Gen.printableChar) ++ Gen.stringBounded(257, 300)(Gen.printableChar)
      check(invalidLongnames) { name =>
        assertTrue(
          Longname.from(name) == Validation.fail(ValidationException("Longname must be 3 to 256 characters long."))
        )
      }
    },
    test("pass a valid value and successfully create value object") {
      val validLongnames = Gen.stringBounded(3, 256)(Gen.printableChar)
      check(validLongnames) { name =>
```

seakayone replied:
I have added d60a54f7aadc9b475bf4f0b9100799436f1cf785 to the other PR, which should clarify the naming a bit.

mpro7 reviewed (approved):
Looks good, apart of places where I suggest changes.

Also general question, why to not keep store error messages in object vals instead of keep repeating them?

seakayone commented:
> Also general question, why to not keep store error messages in object vals instead of keep repeating them?

The error message is only used once in production code and is encapsulated in the function returning the error. This approach keeps it close to the actual usage of the code responsible for validation.

I have introduced redundancy in the test to ensure that a change in the production code will cause the test to fail if we unintentionally alter the desired return value. Previously, one could change any of the `Project.ErrorMessages` in the production code to anything, and the test would always pass because the did not test the actual desired return value but rather that the exact same, changed `String` was returned.
